ui/app/components/secret-edit-toolbar.js

```diff
@@ -32,80 +30,46 @@
  * @param {boolean} isV2 - KV type
  * @param {boolean} isWriteWithoutRead - boolean describing permissions
  * @param {boolean} secretDataIsAdvanced - used to determine if show JSON toggle
- * @param {boolean} showAdvacnedMode - used for JSON toggle
+ * @param {boolean} showAdvancedMode - used for JSON toggle
  * @param {object} modelForData - a modified version of the model with secret data
  * @param {boolean} canUpdateSecretData - permissions that show the create new version button or not.
- * @param {string} codemirrorString - used to copy the JSON
- * @param {object} wrappedData - when copy the data it's the token of the secret returned.
  * @param {object} editActions - actions passed from parent to child
  */
 /* eslint ember/no-computed-properties-in-native-classes: 'warn' */
 import Component from '@glimmer/component';
 import { action } from '@ember/object';
-import { not } from '@ember/object/computed';
 import { inject as service } from '@ember/service';
 import { tracked } from '@glimmer/tracking';
+import { task } from 'ember-concurrency';
+import { waitFor } from '@ember/test-waiters';
 
 export default class SecretEditToolbar extends Component {
   @service store;
   @service flashMessages;
 
   @tracked wrappedData = null;
-  @tracked isWrapping = false;
-  @not('wrappedData') showWrapButton;
 
   @action
   clearWrappedData() {
     this.wrappedData = null;
   }
 
-  @action
-  handleCopyError() {
-    this.flashMessages.danger('Could Not Copy Wrapped Data');
-    this.send('clearWrappedData');
-  }
-
-  @action
-  handleCopySuccess() {
-    this.flashMessages.success('Copied Wrapped Data!');
-    this.send('clearWrappedData');
-  }
-
-  @action
-  handleWrapClick() {
-    this.isWrapping = true;
-    if (this.args.isV2) {
-      this.store
-        .adapterFor('secret-v2-version')
-        .queryRecord(this.args.modelForData.id, { wrapTTL: 1800 })
-        .then((resp) => {
-          this.wrappedData = resp.wrap_info.token;
-          this.flashMessages.success('Secret Successfully Wrapped!');
-        })
-        .catch(() => {
-          this.flashMessages.danger('Could Not Wrap Secret');
-        })
-        .finally(() => {
-          this.isWrapping = false;
-        });
-    } else {
-      this.store
-        .adapterFor('secret')
-        .queryRecord(null, null, {
-          backend: this.args.model.backend,
-          id: this.args.modelForData.id,
-          wrapTTL: 1800,
-        })
-        .then((resp) => {
-          this.wrappedData = resp.wrap_info.token;
-          this.flashMessages.success('Secret Successfully Wrapped!');
-        })
-        .catch(() => {
-          this.flashMessages.danger('Could Not Wrap Secret');
-        })
-        .finally(() => {
-          this.isWrapping = false;
-        });
-    }
+  @task
+  @waitFor
+  *handleWrapClick() {
+    yield this.store
```

Comment:
It's a little strange to me. to combine `yield` (which can be used like async/await) and promise style (`.then().catch`). I would suggest changing to something like: 
```
try {
  const resp = yield this.store...
  this.wrappedData = resp.wrap_info.token;
  this.flashMessages.success('Secret successfully wrapped!');
} catch () {
  this.flashMessages.danger('Could not wrap secret.');
}
```

Reply:
I copied this from the original component - I can update to match the new one!